Code/Framework/AzCore/AzCore/RTTI/TypeSafeIntegral.h

```diff
@@ -32,6 +32,24 @@ namespace AZStd
     AZ_DEFINE_ENUM_RELATIONAL_OPERATORS(TYPE_NAME)                                                                                                  \
     AZ_DEFINE_ENUM_BITWISE_OPERATORS(TYPE_NAME)
 
+//! This implements AZStd to_string methods for a type safe integral alias.
+//! This must be placed in global scope (not in a namespace) and the TYPE_NAME should be fully qualified.
+//! Usage: AZ_TYPE_SAFE_INTEGRAL_TOSTRING(TypeSafeClassName, int8_t/uint8_t/int16_t/uint16_t/int32_t/uint32_t/int64_t/uint64_t);
+#define AZ_TYPE_SAFE_INTEGRAL_TOSTRING(TYPE_NAME, BASE_NAME)                                                                               \
+    namespace AZStd                                                                                                                        \
+    {                                                                                                                                      \
+        inline void to_string(AZStd::string& str, const TYPE_NAME& value)                                                                  \
```

Comment:
Actually this one should be templated to work with other string types(char array, fixed_string, etc...).

You can see the conversions.h file for an [example](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/std/string/conversions.h#L264-L270).

```
    template<class Str>
    void to_string(Str& str, const TYPE_NAME value)
    {
        to_string(str, AZStd::to_underlying(value));
    }
```

Reply:
Ah great, will update.